src/towncrier/_project.py

```diff
@@ -51,7 +52,7 @@ def get_version(package_dir: str, package: str) -> str:
         return version.strip()
 
     if isinstance(version, Version):
-        return version.base().strip()
+        return cast(str, version.base().strip())
```

Comment:
I guess that this is more related to https://github.com/twisted/towncrier/pull/431 

but we can have it here, no problem.